### PR TITLE
Guide adjustments

### PR DESCRIPTION
**Pull request by Vechro**
This PR makes the following changes to the website:

* Adds Discord to the list of Mint communities
* Removes Spectrum from the list of Mint communities
* Fixes a sizeable amount of typos and other errors in the guide
* Fixes the incorrect viewport meta tag values (Chromium complains about these in the console on the current site)
* Adds a small new page about the `next` keyword
* `for` page has been renamed to `for...of` as it *could* otherwise imply it's a regular for-loop with an index

### CODE REVIEW

File: app/views/guides/pages/getting-started/handling-events.haml [hunk 1]
```diff
@@ -19,10 +19,14 @@
 
   An event attribute is always starts with
   %code on
-  followed by the name of the event (capitalized), for example:
+  followed by the name of the event (
+  %a(href="/guide/reference/javascript-interop/decode-expression")
```
Comment (gdotdesign) [inline app/views/guides/pages/getting-started/handling-events.haml:23]: Why does this link to the decode expression?

File: app/views/guides/pages/reference/control-expressions/next.haml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+%p
+  %code next
+  is a very straightforward expression. It allows you to
+  %b locally
+  modify the `state` variables in
+  %a(href="/guide/reference/components/internal-state") Components
+  and
+  %a(href="/guide/reference/stores") Stores
+  . For more information on usage you should look at the examples on the linked pages.
```
Comment (gdotdesign) [inline app/views/guides/pages/reference/control-expressions/next.haml:13]: No new line at the end of the file.

File: app/views/icons/_discord.html [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 245 240">
+  <path
+    d="M104.4 103.9c-5.7 0-10.2 5-10.2 11.1s4.6 11.1 10.2 11.1c5.7 0 10.2-5 10.2-11.1.1-6.1-4.5-11.1-10.2-11.1zM140.9 103.9c-5.7 0-10.2 5-10.2 11.1s4.6 11.1 10.2 11.1c5.7 0 10.2-5 10.2-11.1s-4.5-11.1-10.2-11.1z"
+    fill="black" />
+  <path
+    d="M189.5 20h-134C44.2 20 35 29.2 35 40.6v135.2c0 11.4 9.2 20.6 20.5 20.6h113.4l-5.3-18.5 12.8 11.9 12.1 11.2 21.5 19V40.6c0-11.4-9.2-20.6-20.5-20.6zm-38.6 130.6s-3.6-4.3-6.6-8.1c13.1-3.7 18.1-11.9 18.1-11.9-4.1 2.7-8 4.6-11.5 5.9-5 2.1-9.8 3.5-14.5 4.3-9.6 1.8-18.4 1.3-25.9-.1-5.7-1.1-10.6-2.7-14.7-4.3-2.3-.9-4.8-2-7.3-3.4-.3-.2-.6-.3-.9-.5-.2-.1-.3-.2-.4-.3-1.8-1-2.8-1.7-2.8-1.7s4.8 8 17.5 11.8c-3 3.8-6.7 8.3-6.7 8.3-22.1-.7-30.5-15.2-30.5-15.2 0-32.2 14.4-58.3 14.4-58.3 14.4-10.8 28.1-10.5 28.1-10.5l1 1.2c-18 5.2-26.3 13.1-26.3 13.1s2.2-1.2 5.9-2.9c10.7-4.7 19.2-6 22.7-6.3.6-.1 1.1-.2 1.7-.2 6.1-.8 13-1 20.2-.2 9.5 1.1 19.7 3.9 30.1 9.6 0 0-7.9-7.5-24.9-12.7l1.4-1.6s13.7-.3 28.1 10.5c0 0 14.4 26.1 14.4 58.3 0 0-8.5 14.5-30.6 15.2z"
+    fill="black" />
+</svg>
```
Comment (gdotdesign) [inline app/views/icons/_discord.html:8]: No new line at the end of the file.

File: app/views/guides/pages/getting-started/handling-events.haml (hunk 1 above)
Comment (Vechro) [reply app/views/guides/pages/getting-started/handling-events.haml]: Was supposed to link to the camelCase article on wikipedia, fixed

**Pull request merged**